lenskit/algorithms/als.py

```diff
@@ -466,8 +492,16 @@ def _train_iters(self, current, uctx, ictx):
             yield current
 
     def predict_for_user(self, user, items, ratings=None):
-        # look up user index
-        return self.score_by_ids(user, items)
+        if ratings is not None:
+            # get the indexes for the items in ratings
+            indexes = []
+            for i in ratings.index: # ratings has as index the item id
+                indexes.append(self.item_index_.get_loc(i))
+            user_feature = _train_bias_row_lu(indexes, ratings.values, self.item_features_, self.regularization)
```

Comment:
Here is the problem. Rows are trained on _normalized_ ratings - you need to subtract the bias out of the ratings before training the row.

Reply:
You can also make this more efficient, and need to deal with an issue where a rated item isn't in the item index.

Quick way to do this:

```python
ri_idxes = self.item_index_.get_indexer_for(ratings.index.values)
ri_good = ri_idxes >= 0
ri_items = ri_idxes[ri_good]
ri_values = ratings.values[ri_good]
```

Then you can pass `ri_items` and `ri_values` to `_train_bias_row_lu`.